apps/condo/domains/meter/utils/serverSchema/index.js

```diff
@@ -67,6 +70,61 @@ const getAvailableResidentMeters = async (userId) => {
     })
 }
 
+/**
+ * Get all meter report periods, which resident has access to,
+ * Mostly used in access, that's why used native keystone utils
+ * @param userId - id of user
+ * @returns {Promise<Array<unknown>>} list of meters ids which are available for resident
+ */
+const getAvailableResidentMeterReportPeriods = async (userId) => {
+    const userResidents = await find('Resident', {
+        user: { id: userId, deletedAt: null },
+        property: { deletedAt: null },
+        organization: { deletedAt: null },
+        deletedAt: null,
+    })
+    const residentIds = userResidents.map(resident => resident.id)
+    const residentsByIds = Object.assign({}, ...userResidents.map(obj => ({ [obj.id]: obj })))
+
+    const userConsumers = await find('ServiceConsumer', {
+        resident: { id_in: residentIds, deletedAt: null },
+        organization: { deletedAt: null },
+        deletedAt: null,
+    })
+
+    const selectionsByOrganization = uniq(userConsumers.map(serviceConsumer => ({
+        organization: { id: get(residentsByIds, [serviceConsumer.resident, 'organization']) },
+        property_is_null: true,
+    })))
+
+    const selectionsByProperty = uniq(userConsumers.map(serviceConsumer => ({
+        organization: { id: get(residentsByIds, [serviceConsumer.resident, 'organization']) },
+        property: { id: get(residentsByIds, [serviceConsumer.resident, 'property']) },
+    })))
+
+    const orStatement = selectionsByOrganization.map(selection => ({
+        AND: [
+            selection,
+        ],
+    }))
+
+    selectionsByProperty.map(selection => orStatement.push({
+        AND: [
+            selection,
+        ],
+    }))
```

Comment:
strange code `arr1.map(item => arr2.push(item))`

`map` is used for transformations like `const newArr = oldArr.map(item => someTransform(item))`